packages/synapse-react-client/src/lib/containers/synapse_form_wrapper/WarningDialog.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react'
+import {
+  Box,
+  Button,
+  ButtonProps,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  Stack,
+  SxProps,
+} from '@mui/material'
+import CloseIcon from '@mui/icons-material/Close'
+
+export type WarningDialogProps<T = any> = {
+  title: string
+  content: string | JSX.Element
+  confirmButtonText?: string
+  className?: string
+  open: boolean
+  onConfirm: (...args: T[]) => unknown
+  onConfirmCallbackArgs?: Parameters<WarningDialogProps['onConfirm']>
+  confirmButtonColor?: ButtonProps['color']
+  onCancel: () => void
+}
+
+export type CloseButtonProps = {
+  sx?: SxProps
+  onClick?: () => void
+}
+
+export const CloseButton: React.FC<CloseButtonProps> = ({
+  sx = { color: 'grey.700' },
+  onClick,
+}) => {
+  return (
+    <IconButton sx={sx} onClick={onClick}>
+      <CloseIcon />
+    </IconButton>
+  )
+}
```

Comment:
May be useful for a generic Mui Dialog wrapper component or other components that use Mui Dialogs with a close button